### PR TITLE
[GH-2552] Add ST_OrientedEnvelope

### PR DESCRIPTION
**Pull request by joonaspessi**
## Did you read the Contributor Guide?

Yes, I have read the [Contributor Rules](https://sedona.apache.org/latest/community/rule/) and [Contributor Development Guide](https://sedona.apache.org/latest/community/develop/)

## Is this PR related to a ticket?

Yes, and the PR name follows the format `[SEDONA-XXX] my subject`.

## What changes were proposed in this PR?

Implement `ST_OrientedEnvelope` function by using JTS `org.locationtech.jts.algorithm.MinimumAreaRectangle.getMinimumRectangle()` function. 

## How was this patch tested?

- Java, Scala and Python tests
- Direct Pass through from JTS so not that much sense to go into corner cases

## Did this PR include necessary documentation updates?

- Yes, I am adding a new API. I am using the [current SNAPSHOT version number](https://github.com/apache/sedona/blob/99239524f17389fc4ae9548ea88756f8ea538bb9/pom.xml#L29) in `vX.Y.Z` format.
- Yes, I have updated the documentation.
- No, this PR does not affect any public API so no need to change the documentation.

### CODE REVIEW

File: docs/api/sql/Function.md [hunk 1]
```diff
@@ -3490,6 +3490,26 @@ Output:
 3
 ```
 
+## ST_OrientedEnvelope
+
+Introduction: Returns the minimum-area rotated rectangle enclosing a geometry. The rectangle may be rotated relative to the coordinate axes. Degenerate inputs may result in a Point or LineString being returned.
+
+Format: `ST_OrientedEnvelope(geom: Geometry)`
+
+Since: `v1.8.x`
```
Comment (jiayuasu) [inline docs/api/sql/Function.md]:
```suggestion
Since: `v1.8.1`
```

File: spark/common/src/main/scala/org/apache/spark/sql/sedona_sql/expressions/st_functions.scala [hunk 2]
```diff
@@ -504,6 +504,11 @@ object st_functions {
   def ST_MinimumBoundingRadius(geometry: String): Column =
     wrapExpression[ST_MinimumBoundingRadius](geometry)
 
+  def ST_OrientedEnvelope(geometry: Column): Column =
```
Comment (jiayuasu) [inline spark/common/src/main/scala/org/apache/spark/sql/sedona_sql/expressions/st_functions.scala:507]: Please add a test for this

File: python/sedona/spark/sql/st_functions.py [hunk 3]
```diff
@@ -1595,6 +1595,18 @@ def ST_NumInteriorRing(geometry: ColumnOrName) -> Column:
     return _call_st_function("ST_NumInteriorRing", geometry)
 
 
+@validate_argument_types
+def ST_OrientedEnvelope(geometry: ColumnOrName) -> Column:
```
Comment (jiayuasu) [inline python/sedona/spark/sql/st_functions.py:1599]: please add a test for this.

File: common/src/main/java/org/apache/sedona/common/Functions.java [hunk 4]
```diff
@@ -1227,6 +1228,10 @@ public static Geometry minimumBoundingCircle(Geometry geometry, int quadrantSegm
     return circle;
   }
 
+  public static Geometry orientedEnvelope(Geometry geometry) {
```
Comment (jiayuasu) [inline common/src/main/java/org/apache/sedona/common/Functions.java:1231]: do you mind also porting this to Flink and Snowflake?

File: python/sedona/spark/sql/st_functions.py (hunk 3 above)
Comment (joonaspessi) [reply python/sedona/spark/sql/st_functions.py:1599]: Additional tests added for python

File: spark/common/src/main/scala/org/apache/spark/sql/sedona_sql/expressions/st_functions.scala (hunk 2 above)
Comment (joonaspessi) [reply spark/common/src/main/scala/org/apache/spark/sql/sedona_sql/expressions/st_functions.scala:507]: Additional tests added for scala

File: common/src/main/java/org/apache/sedona/common/Functions.java (hunk 4 above)
Comment (joonaspessi) [reply common/src/main/java/org/apache/sedona/common/Functions.java:1231]: Added Flink and Snowflake implementations

**Review by jiayuasu — APPROVED** [review]
Thank you!
